components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/PlacesHistoryStorageTest.kt

```diff
@@ -139,6 +139,40 @@ class PlacesHistoryStorageTest {
         assertEquals("Mozilla", recordedVisits[0].title)
     }
 
+    @Test
+    fun `store can be used to query top frecent site information`() = runBlocking {
+        val toAdd = listOf(
+            "https://www.example.com/123",
+            "https://www.example.com/123",
+            "https://www.example.com/12345",
+            "https://www.mozilla.com/foo/bar/baz",
+            "https://www.mozilla.com/foo/bar/baz",
+            "https://mozilla.com/a1/b2/c3",
+            "https://news.ycombinator.com/",
+            "https://www.mozilla.com/foo/bar/baz"
+        )
+
+        for (url in toAdd) {
+            history.recordVisit(url, PageVisit(VisitType.LINK, RedirectSource.NOT_A_SOURCE))
+        }
+
+        var infos = history.getTopFrecentSites(0)
+        assertEquals(0, infos.size)
+
+        infos = history.getTopFrecentSites(3)
+        assertEquals(3, infos.size)
+        assertEquals("https://www.mozilla.com/foo/bar/baz", infos[0].url)
+        assertEquals("https://www.example.com/123", infos[1].url)
+
+        infos = history.getTopFrecentSites(5)
```

Comment:
can you also add a test case with more than `toAdd.size` items requested? Like, 100.

Reply:
Added